invision-app/app/tests/reducer.test.js

```diff
@@ -0,0 +1,63 @@
+import {moviesReducer } from '../store/reducer';
+import {
+    FETCH_MOVIES,
+    CREATE_MOVIE,
+    DELETE_MOVIE,
+    UPDATE_MOVIE,
+    SORT_BY,
+    FILTER_BY_GENRE,
+    SHOW_DESCRIPTION_MOVIE,
+    SEARCH_MOVIES
+} from '../store/types';
+import moviesData  from '../model/moviesData.json'
+
+describe('Movie Reducer', () => {
+    const initValue = {
+        movies: [],
+        sortedMovies: [],
+        filterKey: 'All',
+        sortKey: 'release_date'
+      };
+
+      it('should return the initial state', () => {
+        expect(moviesReducer(undefined, {})).toEqual(initValue);
+      });
+
+      it('should handle FETCH_MOVIES', () => {
+        const state = moviesReducer(initValue, { type:FETCH_MOVIES, payload: moviesData });
+        expect(state.movies.length).toBe(8);
+      });
+      it('should handle CREATE_MOVIE', () => {     
+        const state = moviesReducer(initValue, { type:CREATE_MOVIE, payload:{movie: moviesData[0]} });
+        expect(state.movies.length).toBe(1);
+        expect(state.movies[0]).toEqual(moviesData[0]);
+      });
+      it('should handle UPDATE_MOVIE', () => {        
+        var state1={movies:moviesData};
+        moviesData[0].title="test";       
```

Comment:
I'd recommend do not mutate movie but copy it using spread operator and after that use it as a separate movie object to check UPDATE_MOVIE behaviour
```suggestion
        const movie = {...moviesData[0], title: "test"};
```